lib/reporter.js

```diff
@@ -6,21 +6,45 @@ var fs = require('mz/fs');
 var defaultExecutable = path.join(__dirname, '..', 'node_modules', '.bin', 'codeclimate');
 var defaultToken = process.env.CODECLIMATE_REPO_TOKEN;
 
+var getAllFiles = function(dir, name) {
```

Comment:
This function does not use i/o - the whole logic is called statically. To be honest, I won't merge synchronous stuff.
`mz/fs` provides such a nice API to do this without blocking.